duckduckgo/Android
# Add tabs data model

Asana Issue URL: https://app.asana.com/0/361428290920652/574254613727470

## Description
• Refactor `PrivacyMonitorRepository` to be a `TabDataRepository` to support upcoming tabs work. We now use a tbaId to retrieve appropriate live data, thus supporting multiple sessions at once (rather than previous single session)
• Refactor `PrivacyMonitor` to `Site` to better represent meaning now that we are developing tabs. This also matches name used by extensions 

## Steps to Test this PR:
This is a developer task. There should be no noticeable difference. Run the app and ensure everything still holds together.
